### PR TITLE
[CARBONDATA-2649] Add code for caching min/max only for specified columns

### PR DESCRIPTION
**Pull request by manishgupta88**
Things done as part of this PR
1. Supported configuring column for caching min/max in driver
2. Added test cases for query verification for COLUMN_META_CACHE and CACHE_LEVEL properties
3. Handled comments for PR #2454 

This PR is dependent on PR #2454 

 - [ ] Any interfaces changed?
 No
 - [ ] Any backward compatibility impacted?
 No
 - [ ] Document update required?
Yes. Will be done as part of jira-2651
 - [ ] Testing done
In Progress
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5733/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6958/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6964/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6969/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5744/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5726/

File: core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentPropertiesAndSchemaHolder.java [hunk 1]
```diff
@@ -289,20 +293,24 @@ public SegmentProperties getSegmentProperties() {
...
 
-    public CarbonRowSchema[] getBlockSchema() {
-      return SchemaGenerator.createBlockSchema(segmentProperties);
+    public synchronized CarbonRowSchema[] getTaskSummarySchema() {
+      return taskSummarySchema;
     }
 
-    public CarbonRowSchema[] getBlocketSchema() {
-      return SchemaGenerator.createBlockletSchema(segmentProperties);
+    public synchronized void setTaskSummarySchema(CarbonRowSchema[] taskSummarySchema) {
```
Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentPropertiesAndSchemaHolder.java]: I think synchronized is no required here , please check

**Comment by manishgupta88** [comment]
@kumarvishal09 ...handled review comments...kindly review and merge

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6993/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5747/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5773/

**Comment by kumarvishal09** [comment]
Lgtm

**Pull request closed without merging**